src/app/core/services/map-zoom.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { environment } from '../../../environments/environment';
+
+const MAX_ZOOM = 1;
+const MIN_ZOOM = 0.25;
+const DEFAULT_ZOOM_STEP = 0.25;
+const DEFAULT_ZOOM_FACTOR = environment.defaultZoomFactor;
+const ONE_HUNGRED = 100;
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MapZoomService {
+  public readonly zoomFactor$ = new BehaviorSubject<number>(
+    DEFAULT_ZOOM_FACTOR
+  );
+
+  public readonly isMaxZoom$ = new BehaviorSubject<boolean>(
+    DEFAULT_ZOOM_FACTOR === MAX_ZOOM
```

Comment:
Please check, we need move it to utils, and remove code duplication
Also for `DEFAULT_ZOOM_FACTOR === MIN_ZOOM`